src/node/node_string.hpp

```diff
@@ -30,16 +30,35 @@ class String<node::Types> {
   public:
     String(const char* s) : m_str(s) {}
     String(const std::string &s) : m_str(s) {}
-    String(const v8::Local<v8::String> &s) : m_str(*Nan::Utf8String(s)) {}
+    String(const v8::Local<v8::String> &s);
     String(v8::Local<v8::String> &&s) : String(s) {}
 
-    operator std::string() const {
+    operator std::string() const& {
         return m_str;
     }
+    operator std::string() && {
+        return std::move(m_str);
+    }
     operator v8::Local<v8::String>() const {
         return Nan::New(m_str).ToLocalChecked();
     }
 };
 
+inline String<node::Types>::String(const v8::Local<v8::String> &s) {
+    if (s.IsEmpty() || s->Length() == 0) {
+        return;
+    }
+    if (s->IsOneByte()) {
```

Comment:
`ContainsOnlyOneByte` might be safer (but slower)

Reply:
No, it isn't safer. `IsOneByte()` checks the type of the string, while  `ContainsOnlyOneByte()` checks the type of the string, and then if it isn't one-byte then checks the contents of the string to see if it actually could be one-byte. As such, `IsOneByte()` has false negatives, not false positives.

Calling `Utf8Length()` after verifying it's the same thing as `Length()` is just a pointless pessimization.

Reply:
Okay, I see why the test was failing without these changes: internal and external one-byte strings are actually different in v8, and the assumption that `Length() == Utf8Length()` only holds for external one-byte strings.

I think trying to do different things for one-byte and two-byte strings isn't worth it given this.